lib/features/steps/multipart_request_steps.rb

```diff
@@ -0,0 +1,126 @@
+    And the multipart request has a non-empty body
+  )
+end
+
+# Tests the number of fields a multipart request contains.
+#
+# @step_input part_count [Integer] The number of expected fields
+Then('the multipart request has {int} fields') do |part_count|
+  parts = Server.current_request[:body]
+  assert_equal(part_count, parts.size)
+end
+
+# Tests the multipart request has at least one field.
+Then('the multipart request has a non-empty body') do
+  parts = Server.current_request[:body]
+  assert(parts.size.positive?, "Multipart request payload contained #{parts.size} fields")
+end
+
+# Tests that a multipart request field exists and is not null.
+#
+# @step_input part_key [String] The key to the multipart element
+Then('the multipart field {string} is not null') do |part_key|
+  parts = Server.current_request[:body]
+  assert_not_nil(parts[part_key], "The field '#{part_key}' should not be null")
+end
+
+# (Deprecated) Retained for backwards compatibility
+# Tests that a multipart request field exists and is not null.
+#
+# @step_input part_key [String] The key to the multipart element
+Then('the field {string} for multipart request is not null') do |part_key|
+  step %(the multipart field "#{part_key}" is not null)
+end
+
+# Tests that a multipart request field equals a string.
+#
+# @step_input part_key [String] The key to the multipart element
+# @step_input expected_value [String] The string to match against
+Then('the multipart field {string} equals {string}') do |part_key, expected_value|
+  parts = Server.current_request[:body]
+  assert_equal(parts[part_key], expected_value)
+end
+
+# (Deprecated) Retained for backwards compatibility
+# Tests that a multipart request field equals a string.
+#
+# @step_input part_key [String] The key to the multipart element
+# @step_input expected_value [String] The string to match against
+Then('the field {string} for multipart request equals {string}') do |part_key, expected_value|
+  step %(the multipart field "#{part_key}" equals "#{expected_value}")
+end
+
+# Tests that a multipart request field is null.
+#
+# @step_input part_key [String] The key to the multipart element
+Then('the multipart field {string} is null') do |part_key|
+  parts = Server.current_request[:body]
+  assert_nil(parts[part_key], "The field '#{part_key}' should be null")
+end
+
+# (Deprecated) Retained for backwards compatibility
+# Tests that a multipart request field is null.
+#
+# @step_input part_key [String] The key to the multipart element
+Then('the field {string} for multipart request is null') do |part_key|
+  step %(the multipart field "#{part_key}" is null)
+end
+
+# Tests that the multipart payload body does not match a JSON fixture.
+# JSON formatted multipart fields will be parsed into hashes.
+#
+# @step_input fixture_path [String] Path to a JSON fixture
+Then('the multipart body does not match the JSON fixture in {string}') do |fixture_path|
```

Comment:
A check for the file existing would be good, so there's a better error if the path is wrong

```suggestion
Then('the multipart body does not match the JSON fixture in {string}') do |fixture_path|
  assert_true(File.exists?(fixture_path, "'#{fixture_path}' does not exist")
```

It might be even nicer to expand this to an absolute path, so it's super clear what's going on:

```suggestion
Then('the multipart body does not match the JSON fixture in {string}') do |fixture_path|
  full_path = File.realpath(fixture_path)
  assert_true(File.exists?(full_path, "'#{full_path}' does not exist")
```